libcanard/canard.c

```diff
@@ -356,14 +346,14 @@ CANARD_PRIVATE int8_t txAVLPriorityPredicate(           //
 /// that are added later always compare greater than their counterparts with the same deadline.
 /// This ensures that CAN frames with the same deadline are, when timed out, dropped in the FIFO order.
 /// Frames that should be dropped earlier compare smaller (i.e., put on the left side of the tree).
-CANARD_PRIVATE int8_t txAVLDeadlinePredicate(           //
-    void* const                        user_reference,  // NOSONAR Cavl API requires pointer to non-const.
+CANARD_PRIVATE ptrdiff_t txAVLDeadlinePredicate(        //
+    const void*                        user_reference,  // NOSONAR Cavl API requires pointer to non-const.
```

Comment:
Missing const qualifier on the user reference. Please check all other occurrences as well.